tensorflow-framework/src/main/java/org/tensorflow/framework/metrics/impl/MetricVariable.java

```diff
@@ -0,0 +1,125 @@
+limitations under the License.
+=======================================================================*/
+package org.tensorflow.framework.metrics.impl;
+
+import org.tensorflow.Operand;
+import org.tensorflow.framework.initializers.Glorot;
+import org.tensorflow.framework.initializers.Initializer;
+import org.tensorflow.framework.initializers.VarianceScaling;
+import org.tensorflow.framework.initializers.Zeros;
+import org.tensorflow.op.Ops;
+import org.tensorflow.op.core.Variable;
+import org.tensorflow.types.family.TFloating;
+import org.tensorflow.types.family.TIntegral;
+import org.tensorflow.types.family.TNumber;
+
+/**
+ * Helper class that holds a metric variable
+ *
+ * @param <U> the data type of the variable
+ */
+public class MetricVariable<U extends TNumber> {
+  private final Variable<U> variable;
+  private final Initializer<U> initializer;
+  private final Ops tf;
+  private boolean initialized;
+
+  /**
+   * Creates a Metric Variable
+   *
+   * @param tf the TensorFlow Ops
+   * @param variable the variable
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variable
+   */
+  public MetricVariable(Ops tf, Variable<U> variable, long seed, Class<U> type) {
+    this(tf, variable, null, seed, type);
+  }
+
+  /**
+   * Creates a Metric Variable
+   *
+   * @param tf the TensorFlow Ops
+   * @param variable the variable
+   * @param initializer the initializer for the variable, if null, then the default for floating
+   *     point types is {@link org.tensorflow.framework.initializers.Glorot} with distribution
+   *     {@link org.tensorflow.framework.initializers.VarianceScaling.Distribution#UNIFORM}, for
+   *     other types the default initializer is {@link org.tensorflow.framework.initializers.Zeros}
+   * @param seed the seed for random number generation. An initializer created with a given seed
+   *     will always produce the same random tensor for a given shape and data type.
+   * @param type the type for the variable
+   * @throws IllegalArgumentException if the type does not inherit from TNumber and the initializer
+   *     is null
+   */
+  @SuppressWarnings("unchecked")
+  public MetricVariable(
+      Ops tf, Variable<U> variable, Initializer<U> initializer, long seed, Class<U> type) {
+    this.tf = tf;
+    this.variable = variable;
+
+    if (initializer == null) {
+      if (TFloating.class.isAssignableFrom(type)) {
+        //noinspection RedundantCast
+        this.initializer =
+            (Initializer<U>) new Glorot<>(tf, VarianceScaling.Distribution.UNIFORM, seed);
+      } else if (TIntegral.class.isAssignableFrom(type)) {
+        this.initializer = new Zeros<>(tf);
+      } else {
+        throw new IllegalArgumentException(
+            String.format("Type %s is not supported for metric variables", type.getSimpleName()));
```

Comment:
Metric Variables have to be a `TNumber`. With Java generics restricting the type to `<U extends TNumber>`, this should not happen. But, I think we should leave it, just in case. This particular code restricts the default initializer, if none is provided.

Reply:
👍 done (by deleting the file)